server.go

```diff
@@ -936,3 +756,280 @@ func (s *Server) handlerHistory(w http.ResponseWriter, r *http.Request) {
+		}
+
+		sourceAPI := jikan{id: id, l: s.l, excludedTypes: bannedTypes, maxEpisodes: maxEpisodes}
+
+		// Request data from API
+		results, err := getMovieData(&sourceAPI)
+
+		if err != nil {
+			s.l.Error("Error while accessing Jikan API: %v", err)
+			data.ErrorMessage = append(data.ErrorMessage, err.Error())
+			return nil, err
+		}
+
+		return results, nil
+	}
+}
+
+func (s *Server) handleTmdb(data *dataAddMovie, w http.ResponseWriter, r *http.Request, sourcelink string) ([]string, error) {
+
+	tmdbEnabled, err := s.data.GetCfgBool("TmdbEnabled", DefaultTmdbEnabled)
+	if err != nil {
+		s.l.Error("Error while retriving config value 'TmdbEnabled':\n %v", err)
+		data.ErrorMessage = append(data.ErrorMessage, "Something went wrong :C")
+		return nil, err
+	}
+
+	if !tmdbEnabled {
+		s.l.Debug("Aborting Tmdb autofill since it is not enabled")
+		data.ErrorMessage = append(data.ErrorMessage, "Tmdb API usage was not enabled by the site administrator")
+		return nil, fmt.Errorf("Tmdb not enabled")
+	} else {
+		// Retrieve token from database
+		token, err := s.data.GetCfgString("TmdbToken", "")
+		if err != nil || token == "" {
+
+			s.l.Debug("Aborting Tmdb autofill since no token was found")
+			data.ErrorMessage = append(data.ErrorMessage, "TmdbToken is either empty or not set in the admin config")
+			return nil, fmt.Errorf("TmdbToken is either empty or not set in the admin config")
+		}
+		// get the movie id
+		rgx := regexp.MustCompile(`[htp]{4}s?:\/\/[^\/]*\/title\/(tt[0-9]*)`)
+		match := rgx.FindStringSubmatch(sourcelink)
+		var id string
+		if len(match) < 2 {
+			s.l.Debug("Regex match didn't find the movie id in %v", sourcelink)
+			data.ErrorMessage = append(data.ErrorMessage, "Could not retrive movie id from provided link")
+			data.ErrLinks = true
+			return nil, fmt.Errorf("Could not retrive movie id from link")
+		} else {
+			id = match[1]
+		}
+
+		sourceAPI := tmdb{id: id, token: token, l: s.l}
+
+		// Request data from API
+		results, err := getMovieData(&sourceAPI)
+
+		if err != nil {
+			s.l.Error("Error while accessing Tmdb API: %v", err)
+			data.ErrorMessage = append(data.ErrorMessage, err.Error())
+			return nil, err
+		}
+
+		return results, nil
+	}
+}
+
+func (s *Server) handleFormfill(data *dataAddMovie, w http.ResponseWriter, r *http.Request) (results []string, links []string) {
+	// Get all needed values from the form
+
+	// Get all links from the corresponding input field
+	linktext := strings.ReplaceAll(r.FormValue("Links"), "\r", "")
+	data.ValLinks = linktext
+
+	// Get the remarks from the corresponding input field
+	remarkstext := strings.ReplaceAll(r.FormValue("Remarks"), "\r", "")
+	data.ValRemarks = remarkstext
+
+	// Check link maxlength
+	maxLinkLength, err := s.data.GetCfgInt(ConfigMaxLinkLength, DefaultMaxLinkLength)
+	if err != nil {
+		s.l.Error("Unable to get %q: %v", ConfigMaxLinkLength, err)
+		s.doError(
+			http.StatusInternalServerError,
+			"something went wrong :C",
+			w, r)
+		return
```

Comment:
Please use explicit returns instead of implicit ones (ie, `return nil, nil`).  Same thing for the other returns in this function.

Reply:
Yeah my bad, will be changed in the next commit